server/serializer/page_subscription.go

```diff
@@ -14,48 +13,93 @@ import (
 type PageSubscription struct {
 	PageID string `json:"pageID"`
 	BaseSubscription
+	UserID string
 }
 
-func (ps PageSubscription) Add(s *Subscriptions) {
+type OldSubscriptionForPage struct {
+	PageSubscription
+}
+
+type OldPageSubscription struct {
+	OldSubscription OldSubscriptionForPage `json:"oldSubscription"`
+}
+
+func (ps *PageSubscription) Add(s *Subscriptions) {
 	if _, valid := s.ByChannelID[ps.ChannelID]; !valid {
 		s.ByChannelID[ps.ChannelID] = make(StringSubscription)
 	}
 	s.ByChannelID[ps.ChannelID][ps.Alias] = ps
 	key := store.GetURLPageIDCombinationKey(ps.BaseURL, ps.PageID)
-	if _, ok := s.ByURLPagID[key]; !ok {
-		s.ByURLPagID[key] = make(map[string][]string)
+	if _, ok := s.ByURLPageID[key]; !ok {
+		s.ByURLPageID[key] = make(map[string]StringArrayMap)
+	}
+	if _, ok := s.ByURLPageID[key][ps.ChannelID]; !ok {
+		s.ByURLPageID[key][ps.ChannelID] = make(map[string][]string)
+	}
+
+	s.ByURLPageID[key][ps.ChannelID][ps.UserID] = ps.Events
+
+	if s.BySpaceID == nil {
+		s.BySpaceID = make(map[string]string)
 	}
-	s.ByURLPagID[key][ps.ChannelID] = ps.Events
 }
 
-func (ps PageSubscription) Remove(s *Subscriptions) {
+func (ps *PageSubscription) Remove(s *Subscriptions) {
 	delete(s.ByChannelID[ps.ChannelID], ps.Alias)
 	key := store.GetURLPageIDCombinationKey(ps.BaseURL, ps.PageID)
-	delete(s.ByURLPagID[key], ps.ChannelID)
+	delete(s.ByURLPageID[key], ps.ChannelID)
 }
 
-func (ps PageSubscription) Edit(s *Subscriptions) {
+func (ps *PageSubscription) GetChannelID() string {
+	return ps.BaseSubscription.ChannelID
+}
+
+func (ps *PageSubscription) Edit(s *Subscriptions) {
 	ps.Remove(s)
 	ps.Add(s)
 }
 
-func (ps PageSubscription) Name() string {
+func (ps *PageSubscription) Name() string {
 	return SubscriptionTypePage
 }
 
-func (ps PageSubscription) GetAlias() string {
+func (ps *PageSubscription) GetAlias() string {
 	return ps.Alias
 }
 
-func (ps PageSubscription) GetFormattedSubscription() string {
+func (ps *PageSubscription) GetUserID() string {
+	return ps.UserID
+}
+
+func (ps *PageSubscription) GetConfluenceURL() string {
+	return ps.GetSubscription().BaseURL
+}
+
+func (ps *PageSubscription) GetEvents() []string {
+	return ps.Events
+}
+
+func (ps *PageSubscription) GetSpaceKeyOrPageID() string {
+	return ps.PageID
+}
+
+func (ps *PageSubscription) GetFormattedSubscription() string {
 	var events []string
 	for _, event := range ps.Events {
 		events = append(events, eventDisplayName[event])
 	}
 	return fmt.Sprintf("\n|%s|%s|%s|%s|", ps.Alias, ps.BaseURL, ps.PageID, strings.Join(events, ", "))
 }
 
-func (ps PageSubscription) IsValid() error {
+func (ps *PageSubscription) GetOldFormattedSubscription() string {
+	var events []string
+	for _, event := range ps.Events {
+		events = append(events, eventDisplayName[event])
+	}
+	return fmt.Sprintf("\n|%s|%s|%s|%s|%s|", ps.Alias, ps.BaseURL, ps.PageID, ps.ChannelID, strings.Join(events, ", "))
```

Comment:
Here too, also in case we remove it/change it in the future and identify regressions.